modules/test_infrastructure/main.tf

```diff
@@ -0,0 +1,142 @@
+}
+
+# https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/main/modules/vnet/README.md
+module "vnet" {
+  source = "../vnet"
+
+  for_each = var.vnets
+
+  name                   = each.value.name
+  create_virtual_network = each.value.create_virtual_network
+  resource_group_name    = coalesce(each.value.resource_group_name, local.resource_group.name)
+  location               = var.location
+
+  address_space = each.value.address_space
+
+  create_subnets = each.value.create_subnets
+  subnets        = each.value.subnets
+
+  network_security_groups = each.value.network_security_groups
+  route_tables            = each.value.route_tables
+
+  tags = var.tags
+}
+
+# https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/main/modules/vnet_peering/README.md
+module "vnet_peering" {
+  source   = "../vnet_peering"
+  for_each = { for k, v in var.vnets : k => v if v.hub_vnet_name != null }
+
+
+  local_peer_config = {
+    name                = "peer-${each.value.name}-to-${each.value.hub_vnet_name}"
+    resource_group_name = local.resource_group.name
+    vnet_name           = each.value.name
+  }
+  remote_peer_config = {
+    name                = "peer-${each.value.hub_vnet_name}-to-${each.value.name}"
+    resource_group_name = try(each.value.hub_resource_group_name, local.resource_group.name)
+    vnet_name           = each.value.hub_vnet_name
+  }
+
+  depends_on = [module.vnet]
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_interface
+resource "azurerm_network_interface" "vm" {
+  for_each = var.test_vms
+
+  name                = each.value.interface_name
+  location            = var.location
+  resource_group_name = local.resource_group.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = module.vnet[each.value.vnet_key].subnet_ids[each.value.subnet_key]
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+locals {
+  password = sensitive(var.test_vm_authentication.password)
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine
+resource "azurerm_linux_virtual_machine" "this" {
+  for_each = var.test_vms
+
+  # checkov:skip=CKV_AZURE_178:This is a test, non-production VM
+  # checkov:skip=CKV_AZURE_149:This is a test, non-production VM
+
+  name                            = each.value.name
+  resource_group_name             = local.resource_group.name
+  location                        = var.location
+  size                            = each.value.size
+  admin_username                  = var.test_vm_authentication.username
+  admin_password                  = local.password
+  disable_password_authentication = false
+  network_interface_ids           = [azurerm_network_interface.vm[each.key].id]
+  allow_extension_operations      = false
+
```

Comment:
What do you think about adding `custom_data` as we have in GWLB example: https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/19d7e939f7de0bbc8038ecabaead9382807dec8a/examples/gwlb_with_vmseries/example.tfvars#L271-L278 in in os profile https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/19d7e939f7de0bbc8038ecabaead9382807dec8a/modules/virtual_machine/main.tf#L93-L97 ? 

It can be optional, but useful if somebody needs to install additional software in test infrastructure.

Reply:
Right, adding this in the upcoming commit.